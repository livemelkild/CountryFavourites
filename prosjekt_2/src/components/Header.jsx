import React from "react"

function Header() {
    return(
        <h2>Live sin header</h2>
    )
}

export default Header;
