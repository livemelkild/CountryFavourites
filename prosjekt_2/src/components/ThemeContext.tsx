import React, {useState, createContext} from 'react';

interface ThemeContextProps {
    theme: 'light' | 'dark' | 'colorful';
    setTheme: (val: ('light' | 'dark' | 'colorful')) => void;
}

interface ThemeProps{}

export const ThemeContext = createContext<ThemeContextProps | null>(null);

export const ThemeContextProvider: React.FC<ThemeProps> = ({children}) => {
    const [theme, setTheme] = useState<'light' | 'dark' | 'colorful'>('light');

    return (
        <ThemeContext.Provider value={{theme, setTheme}}>
            {children}
        </ThemeContext.Provider>
    )
}

export default ThemeContextProvider
