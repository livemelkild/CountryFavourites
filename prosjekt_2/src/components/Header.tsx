import React, {useContext} from "react";
import { ThemeContext } from "./ThemeContext";

function Header() {
  function chooseTheme() {
    let chosenTheme;
    //trykke knapp 1
    if(themeContext.theme === 'light'){
      chosenTheme = 'light';
    }
    //trykke knapp 2
    if(themeContext.theme === 'colorful'){
      chosenTheme = 'colorful';
    }
    //trykke knapp 3
    if(themeContext.theme === 'dark'){
      chosenTheme = 'dark';
    }
    return chosenTheme;

  }
  const themeContext = useContext(ThemeContext)!;

    return(
        <header className = {chooseTheme()}><h1>Hvilket land vil du reise til?</h1></header>
    )
}

export default Header;
