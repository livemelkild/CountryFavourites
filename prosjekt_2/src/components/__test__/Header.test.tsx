import React, {useContext} from "react";
import renderer from "react-test-renderer";
import Header from "../Header";
import ThemeContext from "../ThemeContext";


it("renders without crashing", ()=>{
    const tree = renderer.create(<ThemeContext><Header /></ThemeContext>).toJSON();
    expect(tree).toMatchSnapshot();
})