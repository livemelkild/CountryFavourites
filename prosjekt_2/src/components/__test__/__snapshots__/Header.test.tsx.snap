// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<header
  className=" "
>
  <h1>
    Hvilket land vil du reise til?
  </h1>
</header>
`;
